Lib/importlib/_bootstrap_external.py

```diff
@@ -802,7 +802,7 @@ def __init__(self, fullname, path):
         """Cache the module name and the path to the file found by the
         finder."""
         self.name = fullname
-        self.path = path
+        self.path = path.__fspath__() if hasattr(path, '__fspath__') else path
```

Comment:
@brettcannon There are some test about `None`, `{}` in unittest.

None: [test_finder.py](https://github.com/python/cpython/blob/master/Lib/test/test_importlib/extension/test_loader.py#L91)
{}: [test_file_loader.py](https://github.com/python/cpython/blob/master/Lib/test/test_importlib/source/test_file_loader.py#L228)